open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,505 @@
+     *
+     * @param controlParameters Initiation Object containing periodic limits
+     * @return validation response object indicating whether the provided periodic limits are valid
+     */
+    public static JSONObject validatePeriodicLimits(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        // Retrieve the periodic limits from the control parameters
+        Object periodicLimit = controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+        // Check if the control parameter is a valid JSON array
+        if (!isValidJSONArray(periodicLimit)) {
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_PERIODIC_LIMITS,
+                    ErrorConstants.INVALID_PARAMETER_PERIODIC_LIMITS,
+                    ErrorConstants.PATH_PERIOD_LIMIT);
+        }
+
+        // Check if the periodic limits key is present
+        if (controlParameters.containsKey(ConsentExtensionConstants.PERIODIC_LIMITS)) {
+
+            // Retrieve the JSON array of periodic limits
+            JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+            // Iterate through each periodic limit
+            Iterator<Object> periodicLimitIterator = periodicLimits.iterator();
+
+            while (periodicLimitIterator.hasNext()) {
+                JSONObject limit = (JSONObject) periodicLimitIterator.next();
+
+                // Retrieve values for validation
+                Object amount = limit.getAsString(ConsentExtensionConstants.AMOUNT);
+                Object currency = limit.getAsString(ConsentExtensionConstants.CURRENCY);
+                Object periodType = limit.getAsString(ConsentExtensionConstants.PERIOD_TYPE);
+
+                // validate amount
+                if (!validateAmountCurrency((String) amount, (String) currency)) {
+                    return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR,
+                            ErrorConstants.INVALID_PARAMETER_AMOUNT,
+                            ErrorConstants.PATH_MAXIMUM_INDIVIDUAL_AMOUNT);
+                }
+
+//                if (!validateCurrency(currency)) {
+//                    return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR,
+//                            ErrorConstants.INVALID_PARAMETER_AMOUNT,
+//                            ErrorConstants.PATH_MAXIMUM_INDIVIDUAL_AMOUNT);
+//                }
+
+                //validate period alignment
+                if (!ConsentManageUtil.validatePeriodicAlignment(limit)) {
+                    return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                            ErrorConstants.INVALID_PERIOD_ALIGNMENT, ErrorConstants.PATH_PERIOD_ALIGNMENT);
+                }
+
+                //validate period type
+                if (!ConsentManageUtil.validatePeriodicType(limit)) {
```

Comment:
Why do we pass the `limit` value into the validatePeriodicType method? applicable for L315 as well